Ladybird/Android/src/main/java/org/serenityos/ladybird/LadybirdActivity.kt

```diff
@@ -28,16 +31,24 @@ class LadybirdActivity : AppCompatActivity() {
         binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
         setSupportActionBar(binding.toolbar)
+        urlEditText = binding.urlEditText
         view = binding.webView
+        view.onLoadStart = { url: String, _ ->
+            urlEditText.setText(url, TextView.BufferType.EDITABLE)
+        }
+        urlEditText.setOnEditorActionListener { textView: TextView?, i: Int, _: KeyEvent? ->
+            Boolean
+            if (i == EditorInfo.IME_ACTION_GO || i == EditorInfo.IME_ACTION_SEARCH) {
+                view.loadURL(textView!!.text.toString())
+            }
+            false
+        }
         view.initialize(resourceDir)
+        view.loadURL("https://ladybird.dev")
```

Comment:
You should instead extract a URL from the intent that caused your activity to open. Use something like `intent.data` (or `intent.dataString`), and if that's null, then load the default.